lib/internal/url.js

```diff
@@ -1406,14 +1404,21 @@ function getPathFromURL(path) {
 // as this is the only character that won't be percent encoded by
 // default URL percent encoding when pathname is set.
 const percentRegEx = /%/g;
-function getURLFromFilePath(filepath) {
+function pathToFileUrl(filepath) {
   const tmp = new URL('file://');
   if (filepath.includes('%'))
     filepath = filepath.replace(percentRegEx, '%25');
   tmp.pathname = filepath;
   return tmp;
 }
 
+function toPathIfFileUrl(fileUrlOrPath) {
+  if (fileUrlOrPath == null || !fileUrlOrPath[searchParams] ||
```

Comment:
why not `instanceof` URL?

Reply:
This was what was already in place here, so I didn't want to alter it for fear of compatibility changes. We could certainly try instanceof.

Reply:
i'm curious why it wasn't using `instanceof`, would prefer to use that if we expose it to users

Reply:
Largely because of performance. Checking for the existence of the internal symbol is significantly faster than using `instanceof`

Reply:
@jasnell I'm increasingly uncomfortable with stuff like this that is making things more loose and forgeable than needed. Do we have data on why this is faster, I was under the impression that newer V8s were optimizing out simple `instanceof` checks.

CC: @bmeurer